contributors/design-proposals/multi-label-CNAME.md

```diff
@@ -0,0 +1,127 @@
+# Multi-label subdomain support in CNAME records
+
+Author: Laszlo Janosi (@janosi), Gergely Csatari (@CsatariGergely)
+
+Date: April 2017
+
+Status: New
+
+# Goal
+
+Enable the usage of multi-label subdomains in CNAME records via ExternalName Service.
+Example: `my-service.my-cluster` could be mapped to `your-service.your-cluster`,
+while with the current implementation only `my-service` can be mapped.
+
+# Motivation
+DNS CNAME records are specified in [RFC1034](https://www.ietf.org/rfc/rfc1034.txt)
+and clarified in Section 10 of [RFC2181](https://www.ietf.org/rfc/rfc2181.txt).
+These allow the usage of multi-label subdomains in CNAME records, and not only
```

Comment:
Can you provide more use cases to support this?  The concept of a single service label is very deeply embedded in Kubernetes architecture, and it's going to be very expensive to change direction.  I'd like to understand what scenarios you can't achieve today and how you see this functional change enabling new use cases that improve Kubernetes.  Otherwise, it's going to be very difficult.